pkg/client/cli/cmds_intercept.go

```diff
@@ -505,6 +520,37 @@ func (is *interceptState) getMountPoint() (string, bool, error) {
 	return mountPoint, doMount, err
 }
 
+func makeIngressInfo(ingressHost string, ingressPort int32, ingressTLS bool, ingressL5 string) (*manager.IngressInfo, error) {
+	ingress := &manager.IngressInfo{}
+	if hostRx.MatchString(ingressHost) {
+		if ingressPort > 0 {
+			ingress.Host = ingressHost
+			ingress.Port = ingressPort
+			ingress.UseTls = ingressTLS
+
+			if ingress.L5Host == "" { // if L5Host is not present
+				ingress.L5Host = ingressHost
+				return ingress, nil
+			} else { // if L5Host is present
+				if hostRx.MatchString(ingressL5) {
+					ingress.L5Host = ingressL5
+					return ingress, nil
+				} else {
+					return nil, fmt.Errorf("the address provided by --ingress-l5, %s, must match the regex"+
+						" [a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)* (e.g. 'myingress.mynamespace')",
+						ingressL5)
+				}
+			}
+		} else {
```

Comment:
And once the above `else`s have been removed, it becomes very clear that the `if` for this `else` always returns, so this `else` is also redundant.